File: c++/Cutie.cpp
#include "Cutie.h"

Cutie::Cutie()
{
	// Assume no hugs by default to be safe
	this->respondsToHugs = false;
	this->hugs = 0;
	this->tlc = 0;
}

void Cutie::setAcceptsHugs(bool value)
{
	this->respondsToHugs = value;
}

bool Cutie::acceptsHugs()
{
	return this->respondsToHugs;
}

void Cutie::hug()
{
	++hugs;
}

void Cutie::empathy()
{
	++tlc;
}

unsigned int Cutie::showHugs()
{
	return hugs;
}

unsigned int Cutie::showTlc()
{
	return tlc;
}

File: java/Cutie.java
public class Cutie {
    private boolean respondsToHugs;
    private int hugs;
    private int tlc;

    public Cutie() {
        respondsToHugs = false;
        hugs = 0;
        tlc = 0;
    }

    public void setAcceptsHugs(boolean value) {
        respondsToHugs = value;
    }

    public boolean acceptsHugs() {
        return respondsToHugs;
    }

    public void hug() {
        hugs++;
    }

    public void empathy() {
        tlc++;
    }

    public int getNumberOfHugs() {
        return hugs;
    }

    public int getNumberOfTlc() {
        return tlc;
    }
}
